src/components/App/App.jsx

```diff
@@ -0,0 +1,256 @@
+import LoginModal from "../LoginModal/LoginModal";
+import SignUpSuccessModal from "../SignUpSucessModal/SignUpSuccessModal";
+
+import CurrentUserContext from "../../contexts/CurrentUserContext";
+import { signUp, signIn, handleToken } from "../../utils/api";
+import { checkFakeToken } from "../../utils/auth";
+import { getArticles, saveArticles } from "../../utils/NewsAPI";
+
+function App() {
+  const [activeModal, setActiveModal] = useState("");
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [currentUser, setCurrentUser] = useState({});
+  const [searchedNewsArticles, setSearchedNewsArticles] = useState(false);
+  const [newsArticlesSearchedResults, setNewsArticlesSearchedResults] =
+    useState([]);
+  const [savedNewsArticles, setSavedNewsArticles] = useState([]);
+  const [newsArticlesCounts, setNewsArticlesCounts] = useState(0);
+  const [keyword, setKeyword] = useState("");
+  const [keywords, setKeywords] = useState([]);
+
+  const navigate = useNavigate();
+
+
+  function handleSearchNewsArticles(news) {
+    console.log("this is the news", news);
+    setKeyword(news);
+    setKeywords([...keywords, news]);
+
+    setIsLoading(true);
+    getArticles({ keyword: news })
+      .then((res) => {
+        if (!searchedNewsArticles) {
+          setSearchedNewsArticles(true);
+        }
+        setNewsArticlesSearchedResults(res.articles);
+        setNewsArticlesCounts(3);
+      })
+      .catch((err) => {
+        console.error("There are no articles", err);
+        setNewsArticlesSearchedResults([]);
+        setSearchedNewsArticles(true);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }
+
+  function handleSignIn({ email, password, username }) {
+    setIsLoading(true);
+    console.log({ email, password, username });
+    setIsLoggedIn(true);
+    setCurrentUser({ email: email, username: username });
+    setIsLoading(false);
+    closeActiveModal();
+  }
+
+  function handleSignUp({ email, password, username }) {
+    setIsLoading(true);
+    console.log({ email, password, username });
+    setIsLoggedIn(true);
+    setCurrentUser({ email: email, username: username });
+    setIsLoading(false);
+    closeActiveModal();
+  }
+
+  function handleSignOut() {
+    setIsLoggedIn(false);
+    setCurrentUser({});
+    setSavedNewsArticles([]);
+    navigate("/");
+  }
+
+  function handleSavedNewsArticles(article) {
+    if (!currentUser) {
+      return;
+    }
+    console.log(keyword);
+    const checkIfSavedNewsArticles = savedNewsArticles.some(
+      (news) => news.title === article.title
+    );
+    console.log(checkIfSavedNewsArticles);
+    if (checkIfSavedNewsArticles) {
+      return;
+    }
+
+    article.keyword = keyword;
+    saveArticles(article).then(() => {
+      console.log(article);
+      console.log([...savedNewsArticles, article]);
+      setSavedNewsArticles([...savedNewsArticles, article]);
+    });
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere in the code**